models/users.js

```diff
@@ -0,0 +1,211 @@
+'use strict';
+
+var Db        = require('../config/database');
+var Session   = require('./sessions');
+var validator = require('validator');
+var mongoose  = require('mongoose');
+
+// Create the schema for a user
+var UserSchema = mongoose.Schema({
+    username : { type: String, required: true },
+    email    : { type: String, required: true },
+    password : { type: String, required: true }
+});
+
+// Creates a collection named users in MongoDB
+var UserMongoModel = Db.model('users', UserSchema);
+
+// Takes username, password, email and saves a new user to MongoDB
+function createUser(username, password, email, callback) {
+    var newUser = new UserMongoModel({
+        username : username,
+        email    : email,
+        password : password
+    });
+
+    // Check if newUser.username is already in UserMongoModel
+    UserMongoModel.find({ username : username }).limit(1).exec(function(err, user) {
```

Comment:
Hey @tuan91187dsf, I just tested it out and `.find()` returns the whole document, as opposed to just a cursor. This might be because we're not using Mongo directly; instead we're using Mongoose. This means that `.find().limit(1)` behaves exactly like `.findOne()`, so we should go with the latter for simplicity's sake.